### PR TITLE
Leo/sy 685 integration testing

### PR DESCRIPTION
**Pull request by LeonLiur**
# Feature Pull Request Template

## Key Information

- [SY-685](https://linear.app/synnaxlabs/issue/SY-685/integration-testing)

## Description

Integration testing framework, integration tests & GitHub actions CI workflow

## Revealed bugs

https://linear.app/synnaxlabs/issue/SY-1025/[cesium]-server-reading-incorrect-number-of-samples-revealed-by

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added relevant tests to cover the changes.
- [x] I have updated user facing documentation accordingly. - **None added**

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
with necessary manual QA steps to test my change. - **Let's discuss whether the load tests should be part of QA requirements**

## Breaking Changes

Committed `pnpm.lock`, may need particular attention.

Also changed error message for when writers collide with each other and when opening a writer against an existing domain.

### CODE REVIEW

**Review by emilbon99 — CHANGES_REQUESTED** [review]

File: client/ts/examples/node/liveStream.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ const client = new Synnax({
 // We can just specify the names of the channels we'd like to stream from.
 const read_from = [
     "stream_write_example_time",
-    "stream_write_example_data_1",
-    "stream_write_example_data_2"
+    // "stream_write_example_data_1",
```
[the remark this answers is not in the document]
Comment (LeonLiur) [reply client/ts/examples/node/liveStream.js:32]: You're right, no they don't.

File: integration/cluster.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"error in starting cluster.\nstdout: %s\nstderr: %s\n",
+			stdOut.String(),
+			stdErr.String(),
+		), func() error { return nil }
+	}
+
+	time.Sleep(5 * telem.Second.Duration())
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply integration/cluster.go:56]: @LeonLiur We should honestly make connectivity check an insecure API that doesn't require authentication

[the remark this answers is not in the document]
Comment (emilbon99) [reply integration/cluster.go:56]: By the way it needs to be a POST request to work

[the remark this answers is not in the document]
Comment (emilbon99) [reply integration/cluster.go:56]: Can you go ahead and make connectivity check an API that doesn't require auth?

**Comment by Lham42** [comment]
test-integration failing

**Comment by Lham42** [comment]
backmerge main

File: integration/cluster.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lham42) [reply integration/cluster.go:55]: has this been addressed?

[the remark this answers is not in the document]
Comment (LeonLiur) [reply integration/cluster.go:56]: Addressed

**Review by Lham42 — APPROVED** [review]
Approve wants you address comments and make note that we will address the integration tests in another linear issue

**Comment by Lham42** [comment]
Looks like the synnax test failing was just a flaky test. For any issues you created from this pr - can you also link them in the pr descriptions as issues exposed by these changes?

**Review by emilbon99 — APPROVED** [review]

**Comment by LeonLiur** [comment]
> Looks like the synnax test failing was just a flaky test. For any issues you created from this pr - can you also link them in the pr descriptions as issues exposed by these changes?

Yes, done

**Pull request merged**